tls_assets.go

```diff
@@ -76,6 +78,13 @@ var ClusterComponents = []ClusterComponent{
 	ServiceAccountComponent,
 }
 
+// MonitoringComponents is a slice enumerating all the components that make up
+// monitoring.
+var MonitoringComponents = []ClusterComponent{
```

Comment:
I don't want to add monitoring components to `ClusterComponents`, as they're not required for the guest cluster - these are for monitoring systems to access the guest cluster. I do want a list of them though, so we can clean them up properly.

This also avoids issues with `SearchCerts` for older clusters that don't have certificates for guest cluster scraping.